### ISSUE TITLE
issue: react hook form breaks blitz.js/next.js build since 7.20

### ISSUE THREAD

**Issue by JohannesKlauss** [react-hook-form/react-hook-form#7088]
### Version Number

7.20.1

### Codesandbox/Expo snack

https://github.com/JohannesKlauss/blitzjs-rhf-build-error

### Steps to reproduce

1. Check out the repo mentioned above
2. Run `yarn install`
3. Run `yarn run build`

### Expected behaviour

The build completes without error.

I guess this is related to this commit: https://github.com/react-hook-form/react-hook-form/commit/461c7db172f34e28c2439f64b23dfd173191a58d

Not quite sure though. I can confirm that is worked without a problem for version `7.19.5`. Let me know if you need any help!

### Relevant log output

```shell
(node:6594) UnhandledPromiseRejectionWarning: Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js
require() of ES modules is not supported.
require() of /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.cjs.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/package.json.

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1080:13)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js:1:7)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
(Use `node --trace-warnings ...` to show where the warning was created)
(node:6594) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:6594) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
(node:6594) UnhandledPromiseRejectionWarning: Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js
require() of ES modules is not supported.
require() of /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.cjs.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/package.json.

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1080:13)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js:1:7)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
(node:6594) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 2)

> Build error occurred
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js
require() of ES modules is not supported.
require() of /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/dist/index.cjs.js from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.cjs.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/react-hook-form/package.json.

    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1080:13)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/johannesklauss/Documents/Github/_experimentation/blitzjs-rhf-build-error/node_modules/@hookform/resolvers/zod/dist/zod.js:1:7)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14) {
  type: 'NodeError',
  code: 'ERR_REQUIRE_ESM'
}
error Command failed with exit code 1.
```
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by huantrinh1802** [comment]
Same problem. Confirmed it works if I go to node_modules/react-hook-form/package.json and change type: "module" to type: "commonjs"

**Comment by slbls** [comment]
This occurs when using Next.js as well.

**Comment by nfroidure** [comment]
Same here, this commit should have been done in a major version : https://github.com/react-hook-form/react-hook-form/commit/461c7db172f34e28c2439f64b23dfd173191a58d#

**Comment by gauravverma029** [comment]
This occurs when using Next.js as well.

Same issue for me it break whole application

**Comment by JohannesKlauss** [comment]
Please refrain from "same here" comments. This creates a lot of noise with no new information.

**Comment by GitPSj** [comment]
Also ran in this problem with 7.3.9. An also after updating to 7.20.1. The 7.19.5 version also doesn't seem to work. Not sure where the problem lies here. There hasn't been a next.js update on our package.json for some time either

**Comment by bluebill1049** [comment]
I was hoping this would fix nextJS 12 (12.0.4), which was fixed with the latest version https://github.com/react-hook-form/resolvers/issues/271#issuecomment-973734926 This really give me trouble as to how to tackle this problem. if anyone has a better idea, feel free to support with a concrete solution and fix it. I will revert the type module in package.json for now.
